content/documentation/pages/5-feature-guides/1-streams/9-function-composition.md

```diff
@@ -1,9 +1,331 @@
+
+This application defines the following function beans:
+
+```
+	@Bean
+	public Function<String, String> upper() {
+		return value -> value.toUpperCase();
+	}
+
+	@Bean
+	public Function<String, String> concat() {
+		return value -> "Hello "+ value;
+	}
+
+```
+
+#### Building
+
+You can skip this section if you want to register the `maven` or `docker` resource URI of the `http-transformer` with the Spring Cloud Data Flow server
+
+You can download the source code for this application from Github:
+
+If you are using RabbitMQ binder: [http-transformer-with-RabbitMQ-binder](https://github.com/spring-cloud/spring-cloud-dataflow-samples/raw/master/dataflow-website/stream-developer-guides/feature-guides/streams/dist/composed-http-transformer-rabbitmq.zip)
+After downloading and unpacking the source code, you can build the application using maven:
+
+```
+cd composed-http-transformer-kafka
+./mvnw clean install
+```
+
+If you are using Kafka binder: [http-transformer-with-Kafka-binder](https://github.com/spring-cloud/spring-cloud-dataflow-samples/raw/master/dataflow-website/stream-developer-guides/feature-guides/streams/dist/composed-http-transformer-kafka.zip)
+After downloading and unpacking the source code, you can build the application using maven:
+
+```
+cd composed-http-transformer-rabbitmq
+./mvnw clean install
+```
+
+#### Registering the locally built application
+
+Now register `http-transformer` application by using the Data Flow Shell.
+
+**NOTE**
+
+> For the below app register `--uri` option, replace the directory name and path of the artifact with the value appropriate to your system.
+
+```
+app register --name http-transformer --type source --uri file:///>YOUR-SOURCE-CODE>/target/composed-http-transformer-[kafka/rabbitmq]-0.0.1-SNAPSHOT.jar
+```
+
+#### Registering the readily available application
+
+The maven/docker artifacts of the `http-transformer` application are readily available in both the `Kafka` and `RabbitMQ` binders.
+
+Maven artifact with Kafka binder:
+
+```
+app register --name http-transformer --type source --uri maven://io.spring.dataflow.sample:composed-http-transformer-kafka:0.0.1-SNAPSHOT
+```
+
+Maven artifact with RabbitMQ binder:
+
+```
+app register --name http-transformer --type source --uri maven://io.spring.dataflow.sample:composed-http-transformer-rabbitmq:0.0.1-SNAPSHOT
+```
+
+Docker artifact with Kafka binder:
+
+```
+app register --name http-transformer --type source --uri docker://springcloudstream/composed-http-transformer-kafka:0.0.1-SNAPSHOT
+```
+
+Docker artifact with RabbitMQ binder:
+
+```
+app register --name http-transformer --type source --uri docker://springcloudstream/composed-http-transformer-rabbitmq:0.0.1-SNAPSHOT
+```
+
+#### Deploying the stream
+
+Let's create the stream without the explicit `transform` application from the previous example. Instead, use the `http-transform` that has the functional beans named `upper` and `concat` defined.
```

Comment:
> Instead of the using the `transform` application from the previous example, we will use the `http-transform` application that includes the functional beans with the name `upper()` and `concat()`.